src/DocumentDB.ChangeFeedProcessor/PartitionManagement/EqualPartitionsBalancingStrategy.cs

```diff
@@ -60,11 +60,23 @@ public IEnumerable<ILease> SelectLeasesToTake(IEnumerable<ILease> allLeases)
 
             if (expiredLeases.Count > 0)
             {
-                return expiredLeases.Take(partitionsNeededForMe);
+                var leasesToTake = expiredLeases.Take(partitionsNeededForMe).ToList();
+                foreach (var lease in leasesToTake)
+                {
+                    if (lease is ILeaseEx leaseEx)
+                        leaseEx.AcquireReason = this.IsExpired(lease) ? AcquireReason.Expired : AcquireReason.NoOwner;
```

Comment:
We should do that as part of CategorizeLeases. Also please update the trace above to include the reason.